### PR TITLE
Remove the Program Naming flag.

### PR DESCRIPTION
**Pull request by ppedrot**
It was introduced as an immediately deprecated option in Coq 8.16.

- [x] Added **changelog**.
- [x] Added / updated **documentation**.

### CODE REVIEW

File: doc/sphinx/changes.rst [hunk 1]
```diff
@@ -384,7 +384,7 @@ Commands and options
   used in the general case. This can create incompatibilities
   in scripts relying on such autogenerated names. The old
   scheme can be reactivated using the deprecated flag
-  :flag:`Program Naming`
+  ``Program Naming``
```
Comment (Alizter) [inline doc/sphinx/changes.rst:387]: So it can no longer be activated right?

Comment (ppedrot) [reply doc/sphinx/changes.rst:387]: Indeed. But note that this is the changelog, so it should remain there.

Comment (Alizter) [reply doc/sphinx/changes.rst:387]: Ah whoops.

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**